### ISSUE TITLE
Possible Inconsistancy in the way atlas coms are obtained

### ISSUE THREAD

**Issue by madwilliam** [eddyod/pipeline_utility#19]
There seems to be some inconsistencies about how atlas coordinates in physical coordinates were obtained in different files

in `ActiveBrainAtlasAdmin/neuroglancer/atlas.py` the atlas centers are obtained by:

```
atlas_centers = get_atlas_centers(atlas_box_size, atlas_box_scales, ATLAS_RAW_SCALE)
dst_point_set = np.array([atlas_centers[s] for s in structures if s in common_keys]).T   
dst_point_set = np.diag(atlas_box_scales) @ dst_point_set
```

where the line
`dst_point_set = np.array([atlas_centers[s] for s in structures if s in common_keys]).T   ` 
assigns a subset of coms from atlas_centers to dst_point_set

in the function get_atlas_centers:
```
atlas_centers = get_centers_dict('atlas', input_type_id=MANUAL) #gets the raw com in the database

for structure, origin in atlas_centers.items():
    center = atlas_box_center + np.array(origin) * atlas_raw_scale / atlas_box_scales
    atlas_centers[structure] = center
```
so for each com with coordinate (x,y,z)  the total calculation to transfer atlas coordinate to physical coordinate is:
`(com * atlas_raw_scale /  atlas_box_scales + atlas_box_center) * atlas_box_scales`
as 
```
atlas_raw_scale = 10
atlas_box_scales = [10,10,20]
atlas_box_center = [500,500,150]
```
this is essentially:
```
(com * 10 / [10,10,20] + [500,500,150]) * [10,10,20]
```

in Bili's code `pipeline_utility/notebooks/Bili/notebook/alignment-error-visualization.ipynb` : 
```
def get_atlas_centers(...):
    atlas_centers = sqlController.get_centers_dict('Atlas', input_type_id=1, person_id=16)
    for structure, center in atlas_centers.items():
        center = atlas_box_center + np.array(center) * atlas_raw_scale / atlas_box_scales
        atlas_centers[structure] = center
```
where `sqlController.get_centers_dict` is essentially the same as `get_centers_dict` and gets com data from the table layer_data

the transformation of com in Billi's code is essentially:
``` 
(com * atlas_raw_scale /  atlas_box_scales + atlas_box_center) 
(com * 10 / [10,10,20] + [500,500,150])
```
Assuming the version in the atlas is correct
billi's code at this step is missing the final step of multiplication by [10,10,20]

later in the code there is 
```
offset = [brain_coms[brain][s] - atlas_coms[s]]
scale = np.array([10, 10, 20])
dx, dy, dz = (offset * scale).T
```
where the factor of [10,10,20] is recovered
but would only make sense if brains_coms is lacking this factor as well

brain_coms is x,y,z values from the center_of_mass table which is deprecated and I am not sure what coordinate the coms are stored in.
But it doesn't seem to make sense to do this calculation if brains_com is either in physical or atlas

**Comment by madwilliam** [comment]
### Updates: Bili's conversion is correct

Summary plot of the difference in Bili's and my analysis
![image](https://user-images.githubusercontent.com/23340578/123854454-9554e500-d8d3-11eb-8c2d-46d77365a23b.png)


# differences between me and Bili's plot:

### Dk52 coms selection for starting coms

Bili: used Kui's Com for DK52 from texture detection

Me: used Beth's Com for DK52 from manual annotation

### Affine transformation from DK52 to input brain

Bili: Affine transformation calculated using airlab

Me: Affine transformation calculated using simple itk

### Affine transformation from transformed DK52 com to Atlas

Bili: transformation calculated from manually annotated com to atlas

Me: transformation calculated from transformed DK 52 com to atlas

### Landmarks used

Bili: used union of annotated structures of 5 brains 

Me: used the intersection

### Reasons these differences contributed to larger com offset in bili's case:

1. It's possible that the airlab affine transform did not converge
2. my version directed find transformation that align transformed DK52 coms to atlas while bili's version uses transformation from another set of com to atlas.

**Comment by qobilidop** [comment]
> ### Updates: Bili's conversion is correct
> Summary plot of the difference in Bili's and my analysis
> ![image](https://user-images.githubusercontent.com/23340578/123854454-9554e500-d8d3-11eb-8c2d-46d77365a23b.png)
> 
> # differences between me and Bili's plot:
> ### Dk52 coms selection for starting coms
> Bili: used Kui's Com for DK52 from texture detection
> 
> Me: used Beth's Com for DK52 from manual annotation
> 
> ### Affine transformation from DK52 to input brain
> Bili: Affine transformation calculated using airlab
> 
> Me: Affine transformation calculated using simple itk
> 
> ### Affine transformation from transformed DK52 com to Atlas
> Bili: transformation calculated from manually annotated com to atlas
No, this transformation is handled by Ed's pipeline, and it's always based on the COMs given, not another set of COMs. In this case, there is no difference between our processes.
> 
> Me: transformation calculated from transformed DK 52 com to atlas
> 
> ### Landmarks used
> Bili: used union of annotated structures of 5 brains
> 
> Me: used the intersection
> 
> ### Reasons these differences contributed to larger com offset in bili's case:
> 1. It's possible that the airlab affine transform did not converge
> 2. my version directly finds transformation that align transformed DK52 coms to atlas while bili's version uses transformation from another set of com to atlas.

**Comment by madwilliam** [comment]
I see.  And it would be the rotation and translation calculated by `align_point_sets(src, dst, with_scaling=True)` in [Landmark Registration: Analytical](https://nbviewer.jupyter.org/github/eddyod/pipeline_utility/blob/master/notebooks/Bili/notebook/landmark-registration-analytical.ipynb) correct?

**Comment by madwilliam** [comment]
from the code that Ed pointed me to it seems like the last transformation is from Beth's annotation to the atlas, but I haven't found the code that store to person 1 input_type 4 used for plot 3

**Comment by qobilidop** [comment]
> I see. And it would be the rotation and translation calculated by `align_point_sets(src, dst, with_scaling=True)` in [Landmark Registration: Analytical](https://nbviewer.jupyter.org/github/eddyod/pipeline_utility/blob/master/notebooks/Bili/notebook/landmark-registration-analytical.ipynb) correct?

Yeah, but eventually it comes down to the code in the pipeline.

**Comment by qobilidop** [comment]
> from the code that Ed pointed me to it seems like the last transformation is from Beth's annotation to the atlas, but I haven't found the code that store to person 1 input_type 4 used for plot 3

What I remember is that Ed has written some special scripts to do this, not the one in the production pipeline. Maybe @eddyod can clarify this.

**Comment by madwilliam** [comment]
looks like they are the same code
